feat(ui): Let users navigate to project stats thru main list (ER-1600)

EricHasegawa opened the pull request:
[ER-1600](https://getsentry.atlassian.net/jira/software/c/projects/ER/boards/90?modal=detail&selectedIssue=ER-1600&assignee=712020%3A45bb2666-a66d-424e-94cd-55c5c73bdc57) describes a UX problem in which users have to go through a 3-click process to access individual project statistics.

This PR adds a button to the end of each row that when clicked, takes you to that projects statistics, it also moves the settings link into a button at the end of each row.

It looks like this:
<img width="1145" alt="Screenshot 2023-06-06 at 10 48 28 AM" src="https://github.com/getsentry/sentry/assets/26236981/f2530c21-21f6-4c9c-b76e-415e3a0695f6">


https://github.com/getsentry/sentry/assets/26236981/6f1c4420-37d0-417b-92b4-ccbaa76cead5



[ER-1600]: https://getsentry.atlassian.net/browse/ER-1600?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Commit:
feat(ui): Let users navigate to project stats thru main list (ER-1600)

When browsing through the list of projects, users can click an icon
next to the project name that navigates them to that project's
statistics.

Commit:
feat(ui): Users can nav to project stats from main list (ER-1600)

From the usage stats page, users can click a button that navigates
them to that individual projects statistics.

Commit:
ref(ui): Add clarifying comment (ER-1600)

cathteng commented on static/app/views/organizationStats/usageStatsProjects.tsx:
```diff
@@ -19,6 +20,8 @@ import withProjects from 'sentry/utils/withProjects';
 import {UsageSeries} from './types';
 import UsageTable, {CellProject, CellStat, TableStat} from './usageTable';
 
+type Router = InjectedRouter | null | undefined;
```
instead of passing the Router through multiple children components to get it where you want to be, i think you could use `RouteComponentProps`

EricHasegawa replied:
fixed!

cathteng reviewed (commented):
looking good so far :)

we usually name PRs like `fix(stats): <description>` https://develop.sentry.dev/commit-messages/

it would also be good to add a video of the UI in action!

Commit:
ref(ui): Inject router instead of prop drilling (EN-1600)

 Please enter the commit message for your changes. Lines starting

EricHasegawa commented:
> looking good so far :)
> 
> we usually name PRs like `fix(stats): <description>` https://develop.sentry.dev/commit-messages/
> 
> it would also be good to add a video of the UI in action!

Changed title and added a video

nhsiehgit reviewed (approved):
nice

Non-blocking nits

I'm not familiar with the components being used (updateProjects, withSentryRouter) but afaict this looks good 👍

EricHasegawa commented on static/app/views/organizationStats/usageStatsProjects.tsx:
```diff
@@ -270,6 +269,9 @@ class UsageStatsProjects extends AsyncComponent<Props, State> {
         </Cell>
       );
     });
+    const emptyColumn = <CellStat key="extra" />; // For displaying extra buttons
```
The return above is returning within a `map` function so we'd need to do something like 
```
    return [
      {
        key: SortBy.PROJECT,
        title: t('Project'),
        align: 'left',
        direction: getArrowDirection(SortBy.PROJECT),
        onClick: () => this.handleChangeSort(SortBy.PROJECT),
      },
      {
        key: SortBy.TOTAL,
        title: t('Total'),
        align: 'right',
        direction: getArrowDirection(SortBy.TOTAL),
        onClick: () => this.handleChangeSort(SortBy.TOTAL),
      },
      {
        key: SortBy.ACCEPTED,
        title: t('Accepted'),
        align: 'right',
        direction: getArrowDirection(SortBy.ACCEPTED),
        onClick: () => this.handleChangeSort(SortBy.ACCEPTED),
      },
      {
        key: SortBy.FILTERED,
        title: t('Filtered'),
        align: 'right',
        direction: getArrowDirection(SortBy.FILTERED),
        onClick: () => this.handleChangeSort(SortBy.FILTERED),
      },
      {
        key: SortBy.DROPPED,
        title: t('Dropped'),
        align: 'right',
        direction: getArrowDirection(SortBy.DROPPED),
        onClick: () => this.handleChangeSort(SortBy.DROPPED),
      },
    ]
      .map(h => {
        const Cell = h.key === SortBy.PROJECT ? CellProject : CellStat;

        return (
          <Cell key={h.key}>
            <SortLink
              canSort
              title={h.title}
              align={h.align as Alignments}
              direction={h.direction}
              generateSortLink={h.onClick}
            />
          </Cell>
        );
      })
      .concat([<CellStat key="empty" />]); // Extra column for displaying buttons etc.
```

Although slightly lengthier, I think the const then push approach is clearer to future readers, if you'd prefer the `concat` approach I can change it over though! lmk what you think

Commit:
ref(ui): Minor refactors

EricHasegawa commented on static/app/views/organizationStats/usageStatsProjects.tsx:
```diff
@@ -120,7 +120,6 @@ class UsageStatsProjects extends AsyncComponent<Props, State> {
 
   get tableData() {
     const {projectStats} = this.state;
-
```
fixed!
